### ISSUE TITLE
The version number prefix is not consistent with the composer specification

### ISSUE THREAD

**Issue by kayw-geek** [localheinz/json-normalizer#1045]
Currently,  the package will be delete the 'v' prefix of version number but in the [composer doc](https://getcomposer.org/doc/04-schema.md#version) or [schema](https://getcomposer.org/schema.json) the 'v' prefix is allowed, so I want them to keep be consistent, I would like to hear your thoughts on this.

**Comment by fredden** [comment]
That change was introduced to fix a bug described in https://github.com/ergebnis/json-normalizer/issues/1003.

The composer-normalize tool requires a valid `composer.json` file as input, so all of the changes that are made are moving from one valid format to another. An example: the specification allows packages in the `require` section to be listed in any order; this tool puts them into alphabetical order. The whole point of this tool (and other linters like ESLint and PHP_CodeSniffer) is to create a consistent standard so that the cognitive load of switching projects is lower.

> I want them to keep be consistent

Does this tool produce inconsistent results? If so, please can you provide some examples.

**Comment by kayw-geek** [comment]
I don't think it should be deleted when the version number is not a range but a specific value, for example“xxx/file-log”: “V100.509.0” should not be changed to“xxx/file-log”: “100.509.0”

**Comment by kayw-geek** [comment]
So should we change the code like following:
```DIFF
- '{^(|[!<>]=|[~<>^])v(\d+.*)$}',
+ '{^([!<>]=|[~<>^])v(\d+.*)$}',
```

**Comment by kayw-geek** [comment]
I don't think we should modify the `composer/composer` library in order to fix the bug, and their bug has been fixed now
https://github.com/composer/composer/pull/11764

**Comment by fredden** [comment]
> I don't think we should modify the `composer/composer` library in order to fix the bug, and their bug has been fixed now [composer/composer#11764](https://github.com/composer/composer/pull/11764)

What bug are you referring to? This issue (#1045) looks like a feature request to me.
